### PR TITLE
Write stateful logic

### LINKED ISSUE #9: State setup and usable in context of event handlers
- some simple solution needs to exist so state is available to mutators triggered in event handlers
- maybe mutators and state are setup somewhere else and the event handlers have access to some other event emitter that provides events for each mutation
- maybe rendering code needs to be triggered in here

### PR DESCRIPTION
**Pull request by poltak**
- On creation of an instance it sets up an initial state and affords you either resetting it or performing arbitrary mutations. 
- Mutations are afforded by passing in the mutator fn and any arguments it might need.
- It ensures the render logic is triggered upon every state change
- Example usage might look like:

```js
function main() {
    const logic = new StatefulLogic()

    document.addEventListener('keydown', (event) => {
        if (event.key === 'left') {
            logic.mutate(moveBasketLeft, 10)
        } else if (event.key === 'right') {
            logic.mutate(moveBasketRight, 10)
        }
    })
}
```

### CODE REVIEW

File: statefulLogic.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    constructor() {
+        this.resetState()
+    }
+
+    resetState() {
+        this.state = { ...INIT_STATE }
+    }
+
+    mutate(mutatorFn, ...args = []) {
```
Comment (timotiousPrime) [inline statefulLogic.js]: How come there is a `[]` here?

File: stateMutators.js [hunk 2]
```diff
@@ -1,205 +1,197 @@
...
-    this.denominator = denominator;
-    this.value = numerator / denominator;
-    this.id = id;
+export function FallingObject({
+    posX,
+    posY,
+    numerator,
+    denominator,
+    id,
+    width = FALLING_OBJ_SIZE,
```
Comment (timotiousPrime) [inline stateMutators.js:124]: how come both the width and height have the same FALLING_OBJ_SIZE? Is it because we always want it to be a square?

Comment (poltak) [reply stateMutators.js:124]: Yeah that's what I was thinking. This is just set as the default values though, i.e., if you don't explictly set `width` and `height` when creating a `FallingObject`

Reminds me I made a few fixes to the mutators code like this in 0e4745a8395e05c27e5aec7181086bed8911aa15 , if you haven't already seen it. Some comments in that commit message

Comment (timotiousPrime) [reply stateMutators.js:124]: Thank you for that. So many rookie errors from the rookie

**Review by tranhuuphucuong — APPROVED** [review]

**Pull request merged**